libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/Mocks/MockBotFrameworkClient.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Testing.Mocks
+{
+    /// <summary>
+    /// MockBotFrameworkClient mock.
+    /// </summary>
+    public class MockBotFrameworkClient : BotFrameworkClient
+    {
+        /// <inheritdoc/>
+        public override Task<InvokeResponse<T>> PostActivityAsync<T>(string fromBotId, string toBotId, Uri toUrl, Uri serviceUrl, string conversationId, Activity activity, CancellationToken cancellationToken = default)
+        {
+            var responseActivity = activity;
+
+            if (activity.Text == "skill")
+            {
+                responseActivity = new Activity()
+                {
+                    Type = "message",
+                    Text = "This is the skill talking: hello"
+                };
+            }
```

Comment:
I'm not sure about this. This class is not reusable, except for this test.

Reply:
I understand your concern, as I understand it we could address the issue by renaming the middleware classes to something more specific to the test and/or do the same for the message text, right?

Reply:
Yes, maybe `MockSkillBotFrameworkClient.cs`.

Reply:
Change uploaded, thank you!